dask_expr/repartition.py

```diff
@@ -84,7 +84,7 @@ def _simplify_down(self):
     def _simplify_up(self, parent):
         # Reorder with column projection
         if isinstance(parent, Projection):
-            return type(self)(self.frame[parent.columns], *self.operands[1:])
+            return type(self)(self.frame[parent.operand("columns")], *self.operands[1:])
```

Comment:
**Side thought** (nothing to do here): I wonder if it makes sense to avoid doing this if/when `partition_size` is used (once it is implemented).